Dockerfile

```diff
@@ -19,156 +20,330 @@
+    && nvm use default \
+    && npm install --global npm@$NPM_VERSION bower@$BOWER_VERSION yarn@$YARN_VERSION
+
+#------------------------------------------------------------------------
+# GOLANG - Build as a separate component
+FROM build AS gobuild
+
+ARG GO_DEP_VERSION=0.5.4
+ARG GO_VERSION=1.17.3
+ENV GOPATH=/opt/go
+RUN curl -L https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz | tar -C /opt -xz
+ENV PATH=/opt/go/bin:$PATH
+RUN go version
+RUN curl -ksS https://raw.githubusercontent.com/golang/dep/v$GO_DEP_VERSION/install.sh | bash
+RUN echo "add_local_path /opt/go/bin:\$PATH" > /etc/profile.d/go.sh
+
+#------------------------------------------------------------------------
+# HASKELL STACK
+FROM build AS haskellbuild
+
+ARG HASKELL_STACK_VERSION=2.1.3
+RUN curl -ksS https://raw.githubusercontent.com/commercialhaskell/stack/v$HASKELL_STACK_VERSION/etc/scripts/get-stack.sh | bash -s -- -d /usr/bin
+
+#------------------------------------------------------------------------
+# REPO / ANDROID SDK
+FROM build AS androidbuild
+
+ARG ANDROID_CMD_VERSION=7583922
+ENV ANDROID_HOME=/opt/android-sdk
+
+RUN curl -ksS https://storage.googleapis.com/git-repo-downloads/repo > /usr/bin/repo \
+    && chmod a+x /usr/bin/repo
+
+RUN curl -Os https://dl.google.com/android/repository/commandlinetools-linux-${ANDROID_CMD_VERSION}_latest.zip \
+    && unzip -q commandlinetools-linux-${ANDROID_CMD_VERSION}_latest.zip -d $ANDROID_HOME \
+    && rm commandlinetools-linux-${ANDROID_CMD_VERSION}_latest.zip \
+    && PROXY_HOST_AND_PORT=${https_proxy#*://} \
+    && if [ -n "$PROXY_HOST_AND_PORT" ]; then \
+    # While sdkmanager uses HTTPS by default, the proxy type is still called "http".
+    SDK_MANAGER_PROXY_OPTIONS="--proxy=http --proxy_host=${PROXY_HOST_AND_PORT%:*} --proxy_port=${PROXY_HOST_AND_PORT##*:}"; \
+    fi \
+    && yes | $ANDROID_HOME/cmdline-tools/bin/sdkmanager $SDK_MANAGER_PROXY_OPTIONS \
+    --sdk_root=$ANDROID_HOME "platform-tools" "cmdline-tools;latest"
+COPY docker/android.sh /etc/profile.d
+
+#------------------------------------------------------------------------
+# SCANCODE - Build ScanCode as a separate component
+FROM pythonbuild AS scancodebuild
 
-# Set this to the ScanCode version to use.
 ARG SCANCODE_VERSION="30.1.0"
+ENV SCANCODE_URL "https://github.com/nexB/scancode-toolkit/archive/refs/tags/"
+ENV PATH=/opt/python/shims:$PATH
 
-FROM adoptopenjdk:11-jdk-hotspot-focal AS build
+RUN mkdir -p /opt/scancode \
+    && curl -ksSL ${SCANCODE_URL}/v${SCANCODE_VERSION}.tar.gz | tar -C /opt/scancode -xz --strip-components=1 \
+    && cd /opt/scancode \
+    && PYTHON_EXE=python3 ./configure \
+    # Clean up unneeded installed binaries
+    && rm -rf /opt/scancode/thirdparty \
+    # Run scancode once to generate indexes as superuser
+    && /opt/scancode/bin/scancode --version
 
-COPY . /usr/local/src/ort
+#------------------------------------------------------------------------
+# ORT
+FROM nodebuild as ortbuild
```

Comment:
FYI: This could probably be just `build` now as findings the bootstrapped `node` at Kotlin build time has been fixed as part of https://github.com/oss-review-toolkit/ort/pull/5040.